dev/com.ibm.ws.microprofile.reactive.streams.operators/src/com/ibm/ws/microprofile/reactive/streams/WASReactiveStreamsEngineImpl.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.streams;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
+import org.eclipse.microprofile.reactive.streams.CompletionSubscriber;
+import org.eclipse.microprofile.reactive.streams.spi.Graph;
+import org.eclipse.microprofile.reactive.streams.spi.ReactiveStreamsEngine;
+import org.eclipse.microprofile.reactive.streams.spi.UnsupportedStageException;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.reactivestreams.Processor;
+import org.reactivestreams.Publisher;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.lightbend.microprofile.reactive.streams.zerodep.BuiltGraph;
+import com.lightbend.microprofile.reactive.streams.zerodep.ReactiveStreamsEngineImpl;
+
+@Component(name = "com.ibm.ws.microprofile.reactive.streams.ReactiveStreamsEngineImpl", service = { ReactiveStreamsEngine.class }, property = { "service.vendor=IBM" }, immediate = true, configurationPolicy = ConfigurationPolicy.IGNORE)
+public class WASReactiveStreamsEngineImpl extends ReactiveStreamsEngineImpl implements ReactiveStreamsEngine {
+
+    private static final TraceComponent tc = Tr.register(ReactiveStreamsEngineImpl.class);
+
+    private static ReactiveStreamsEngine singleton = null;
+
+    @Reference(target = "(component.name=com.ibm.ws.threading)")
+    private static ExecutorService serverExecutor;
```

Comment:
I don't think you can make a static method an OSGi service reference.

The pattern I've seen most often where you want to make a single service available statically is to set a static variable from an `@Activate` method on the componenet.

Reply:
On the other hand, if you only need static access to the instance for testing, you're better off moving the code that constructs the static instance into test code.

Reply:
I have changed it to    /**
     * Declarative Services method for setting the Liberty executor.
     *
     * @param svc the service
     */
    @Reference(target = "(component.name=com.ibm.ws.threading)")
    protected void setExecutor(ExecutorService svc) {
        serverExecutor = svc;
    }
as taken from elsewhere